src/coreclr/vm/comutilnative.cpp

```diff
@@ -76,8 +76,44 @@ FCIMPL0(VOID, ExceptionNative::PrepareForForeignExceptionRaise)
 }
 FCIMPLEND
 
+void DeepCopyStackTrace(StackTraceArray &stackTrace, StackTraceArray &stackTraceCopy, PTRARRAYREF *pKeepaliveArray, PTRARRAYREF *pKeepaliveArrayCopy)
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_COOPERATIVE;
+    }
+    CONTRACTL_END;
+
+    if (stackTrace.Get() != NULL)
+    {
+        stackTraceCopy.CopyFrom(stackTrace);
+    }
+    else
+    {
+        stackTraceCopy.Set(NULL);
+    }
+
+    if (*pKeepaliveArray != NULL)
+    {
+        // The stack trace object is the keepalive array with its first slot set to the stack trace I1Array.
+        // Get the number of elements in the dynamic methods array
+        unsigned cOrigKeepalive = (*pKeepaliveArray)->GetNumComponents();
+
+        // ..and allocate a new array. This can trigger GC or throw under OOM.
+        *pKeepaliveArrayCopy = (PTRARRAYREF)AllocateObjectArray(cOrigKeepalive, g_pObjectClass);
+
+        // Deepcopy references to the new array we just allocated
+        memmoveGCRefs((*pKeepaliveArrayCopy)->GetDataPtr(), (*pKeepaliveArray)->GetDataPtr(),
+                      cOrigKeepalive * sizeof(Object *));
+
+        (*pKeepaliveArrayCopy)->SetAt(0, (PTRARRAYREF)stackTraceCopy.Get());
+    }
+}
+
 // Given an exception object, this method will extract the stacktrace and dynamic method array and set them up for return to the caller.
-FCIMPL3(VOID, ExceptionNative::GetStackTracesDeepCopy, Object* pExceptionObjectUnsafe, Object **pStackTraceUnsafe, Object **pDynamicMethodsUnsafe);
+FCIMPL2(VOID, ExceptionNative::GetStackTracesDeepCopy, Object* pExceptionObjectUnsafe, Object **pStackTraceUnsafe);
```

Comment:
```suggestion
FCIMPL1(Object *, ExceptionNative::GetStackTracesDeepCopy, Object* pExceptionObjectUnsafe)
```
This can return the value now that there is just a single value to return

Reply:
Updated the suggestion to `FreezeStackTrace` since the method does not actually perform a deep copy in typical case anymore.